test/TestCase.cpp

```diff
@@ -139,3 +138,36 @@ EVMVersionRestrictedTestCase::EVMVersionRestrictedTestCase(std::string const& _f
 	if (!comparisonResult)
 		m_shouldRun = false;
 }
+
+void EVMVersionRestrictedTestCase::processCompileToFlag()
+{
+	std::optional<uint8_t> eofVersion = solidity::test::CommonOptions::get().eofVersion();
+
+	std::string compileToString = m_reader.stringSetting("compileTo", "any");
+	if (compileToString == "any")
+		return;
+
+	// TODO: This is naive implementation because for now we support only one EOF version.
+	if (compileToString == "legacy" && eofVersion.has_value())
+		m_shouldRun = false;
+	else if (compileToString == ">=1")
+	{
+		// Do not run if non-EOF test run
+		if (!eofVersion.has_value())
+			m_shouldRun = false;
+		// Do not run if before prague fork test run
+		else if (solidity::test::CommonOptions::get().evmVersion() < langutil::EVMVersion::prague())
+			m_shouldRun = false;
+	}
+	else if (compileToString == "legacy,>=1" || compileToString == ">=1,legacy")
+		return;
```

Comment:
You could put this branch above, together with `"any"`. Actually, `"any"` is redundant if we have this and I doubt we'll ever use it :)

Reply:
Done